File: AndroidRetrofitAPI/app/src/main/java/com/example/androidretrofitapi/view/MainActivity.java
package com.example.androidretrofitapi.view;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.androidretrofitapi.R;
import com.example.androidretrofitapi.model.ApiMovie;
import com.example.androidretrofitapi.model.Movie;
import com.example.androidretrofitapi.model.ApiClient;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private List<Movie> movies;
    private RecyclerView recyclerView;
    private MovieAdapter movieAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.rv_movies);
        recyclerView.setLayoutManager(new GridLayoutManager(getApplicationContext(), 2));
        showMovies();
    }

    public void showMovies() {
        Call<List<Movie>> call = ApiClient.getClient().create(ApiMovie.class).getMovies();
        call.enqueue(new Callback<List<Movie>>() {
            @Override
            public void onResponse(Call<List<Movie>> call, Response<List<Movie>> response) {
                if (response.isSuccessful()){
                    movies = response.body();
                    MovieAdapter movieAdapter=new MovieAdapter(movies,getApplicationContext());
                    recyclerView.setAdapter(movieAdapter);
                }
            }

            @Override
            public void onFailure(Call<List<Movie>> call, Throwable t) {
                String messageError = MainActivity.this.getApplicationContext().getString(R.string.connection_error);
                Toast.makeText(MainActivity.this, messageError, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: AndroidRetrofitAPI/app/src/main/java/com/example/androidretrofitapi/view/MainActivity.kt
package com.example.androidretrofitapi.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.androidretrofitapi.R
import com.example.androidretrofitapi.databinding.ActivityMainBinding
import com.example.androidretrofitapi.model.vo.Movies
import com.example.androidretrofitapi.viewmodel.MoviesViewModel

class MainActivity : AppCompatActivity() {


    private var moviesViewModel: MoviesViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupBindings(savedInstanceState)
    }
    fun setupBindings(savedInstanceState: Bundle?) {
        var activityMainBinding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)
        moviesViewModel = ViewModelProviders.of(this).get(MoviesViewModel::class.java)
        activityMainBinding.setModel(moviesViewModel)
        setupListUpdate()
    }

    //levantar el recyclerview
    fun setupListUpdate() {
        moviesViewModel?.callMovies()
        moviesViewModel?.getMovies()?.observe(this, Observer {
                movies: List<Movies> ->
            Log.w("MOVIES", movies.get(0).title ?: "No encontro titulo")
            moviesViewModel?.setMoviesInRecyclerAdapter(movies)
        })
    }
}